### PR TITLE
feat(Production): Added static site generation

### PR DESCRIPTION
**Pull request by HHogg**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,10 +37,14 @@
     "babel-preset-stage-2": "^6.24.1",
     "brandwatch-react-auth": "1.0.3",
     "bw-axiom": "^25.8.0",
+    "clean-webpack-plugin": "^0.1.16",
+    "copy-webpack-plugin": "^4.0.1",
     "css-loader": "^0.28.5",
+    "ejs-loader": "^0.3.0",
```
Comment (pmsorhaindo) [inline package.json]: The ejs loader wasn't needed before, was it because we weren't using the `<%` syntax?

File: src/store/configure.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { compose, createStore, applyMiddleware } from 'redux';
+import { routerMiddleware } from 'react-router-redux';
+import { composeWithDevTools } from 'redux-devtools-extension';
+import reducers from './root';
+
+export default (history) => {
+  const middleware = routerMiddleware(history);
+  const finalCompose = __DEVELOPMENT__ ? composeWithDevTools : compose;
```
Comment (pmsorhaindo) [inline src/store/configure.js]: So the dev tools will always on in dev! 👍

**Comment by pmsorhaindo** [comment]
Ignore the comment, the ejs plugin is needed because we're not using that HtmlWebpackPlugin now. Good to go from me.

File: src/static.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+export default (locals) => {
+  const hash = locals.webpackStats.hash;
+  const history = createMemoryHistory();
+  const store = configureStore(history);
+
+  return template({
+    htmlWebpackPlugin: {
+      options: {
+        stylesheet: `/assets/bundle.${hash}.min.css`,
```
Comment (thatguynamedandy) [inline src/static.js]: good call on the `hash` 👌🏻

File: src/static.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          <App />
+        </ConnectedRouter>
+      </Provider>
+    </BrandwatchReactAuth>,
+    document.getElementById('root')
+  );
+}
+
+export default (locals) => {
+  const hash = locals.webpackStats.hash;
```
Comment (thatguynamedandy) [inline src/static.js]: Very minor but you could destructure.

File: webpack.config.js [hunk 5]
```diff
@@ -36,13 +36,13 @@ module.exports = {
...
     new webpack.NamedModulesPlugin(),
+    new HtmlWebpackPlugin({
+      template: 'index.ejs',
+    }),
     new webpack.DefinePlugin({
       __AUTH_AUDIENCE__: `"${process.env.AUTH_AUDIENCE}"`,
       __AUTH_DOMAIN__: `"${process.env.AUTH_DOMAIN}"`,
-    }),
-    new HtmlWebpackPlugin({
-      favicon: 'favicon.ico',
```
Comment (thatguynamedandy) [inline webpack.config.js]: Was the favicon line doing nothing?

File: webpack.static.config.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+const modulesToTranspile = [ 'viziaauth' ];
+
+module.exports = {
+  entry: './src/static.js',
+  module: {
+    rules: [{
+      test: /\.js$/,
+      use: ['babel-loader'],
+      exclude: new RegExp(`node_modules/(?!(${modulesToTranspile.join('|')}))`),
```
Comment (thatguynamedandy) [inline webpack.static.config.js]: At first I thought the regex might be a slight overkill, I tried the long hand approach (see below), and don't particularly like either, just throwing it there anyway.
```js
exclude: [
  /node_modules\/viziaauth'/
  /node_modules\/someothermodule'/
]
```

Comment (HHogg) [reply webpack.static.config.js]: This regex is doing the opposite of that... it's excluding everything in `node_modules` except the `modulesToTranspile` list. Maybe we could go with using `include` instead? 

```js
include: [
  /src/,
  /node_modules\/viziaauth/
]
```

File: webpack.config.js (hunk 5 above)
Comment (HHogg) [reply webpack.config.js]: There's now an assets folder for static resources which is copied to the static build folder. To cut down on the duplicate config between the dev and static webpack configs, I've put the tag in the ejs file.

**Pull request merged**